superset/assets/javascripts/SqlLab/actions.js

```diff
@@ -142,6 +142,25 @@ export function runQuery(query) {
   };
 }
 
+export function cancelQuery(query) {
+  return function (dispatch) {
+    dispatch(stopQuery(query));
+    const cancelQueryUrl = '/superset/stop_query/';
+    const cancelQueryRequest = { client_id: query.id };
+    $.ajax({
+      type: 'POST',
+      dataType: 'json',
+      url: cancelQueryUrl,
+      data: cancelQueryRequest,
+      success(results) {
+        if (!query.runAsync) {
+          dispatch(querySuccess(query, results));
```

Comment:
i'm curious why we want to call `querySuccess` here. wouldn't we want to show feedback that the query has been cancelled? `querySuccess` is used to show the fetched results.